File: src/leetcode/python/medium/IsSubsequence_392.py
class IsSubsequence_392(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if not s: return True
        if not t: return False
        index = -1
        for c in s:
            try:
                index = t.index(c, index + 1)
            except:
                return False
        return True

File: src/leetcode/java/IsSubsequence_392.java
package leetcode.java.medium;

public class IsSubsequence_392 {
  public boolean isSubsequence(String s, String t) {
    if (s.isEmpty()) return true;
    if (t.isEmpty()) return false;
    char[] arr = s.toCharArray();
    int index = -1;
    for (char ch : arr) {
      index = t.indexOf(ch, index + 1);
      if (index < 0) return false;
    }
    return true;
  }
}
